test/functional/mempool_persist.py

```diff
@@ -112,5 +112,17 @@ def run_test(self):
         assert_raises_rpc_error(-1, "Unable to dump mempool to disk", self.nodes[1].savemempool)
         os.remove(mempooldotnew1)
 
+        NUM_TXS_IN_MEMPOOL = 900
+        self.log.debug("Send %d transactions from node0 (to its own address)" % NUM_TXS_IN_MEMPOOL)
+        self.stop_nodes()
+        self.start_node(0)
+        self.nodes[0].generate(12)
+        for i in range(NUM_TXS_IN_MEMPOOL):
+            self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), Decimal("0.00001"))
+        self.stop_nodes()
+        self.start_node(0)
```

Comment:
is this test essentially racing the mempool acceptance?

Reply:
stopping the node the mempool is dumped and starting the node, LoadMempool gets called again, hopefully with enough work ahead to catch the error when calling savemempool on the next line.